### PR TITLE
UI TransactionDisplay

### PR DESCRIPTION
**Pull request by goldsteinsveta**
The first iteration on the message flow card (aka `TransactionDisplay`). 

Screenshot:

Screenshot 2021-04-29 at 20 16 1

Changes:

- more UI components
- Unboxed Main layout
- Tweaked TransactionTypes
- Pulled relevant for transactions (and seemingly actual) data into a sidebar and the message card, Simplified text labels from TransactionStatus

### CODE REVIEW

File: src/components/Icons.tsx
Comment (hbulgarini) [inline src/components/Icons.tsx:1]: just curious: are we going to use this component to show the corresponding chain icon?

File: src/screens/Main.tsx
Comment (hbulgarini) [inline src/screens/Main.tsx:92]: i will miss you intent of dashboard :/

**Review by hbulgarini — APPROVED** [review]
The PR looks pretty good . I don't have anything to point out.
Anyway i would like to have @tomusdrw to approve this PR before it is merged since in contains some texting that i want him to be aware.

File: src/screens/Main.tsx
Comment (goldsteinsveta) [reply src/screens/Main.tsx:92]: it'll be back :) I promise!

File: src/components/Network.tsx
Comment (wirednkod) [inline src/components/Network.tsx:102]: In all the dbsource cases below you must ensure that dbSource exists (add a ? before each dot). If it does not - then that could break the page

File: src/components/TransactionDisplay.tsx
Comment (wirednkod) [inline src/components/TransactionDisplay.tsx:47]: You should always return a component and never a null. 
It is better to make the length step BEFORE rendering the component. That means outside the TransactionDisplay - before it is called... something like: 
```
!steps.length && <TransactionDisplay transaction={transaction} steps={steps} />
```

**Review by wirednkod — APPROVED** [review]

**Review by wirednkod — CHANGES_REQUESTED** [review]
as per comments

File: src/components/Network.tsx
Comment (hbulgarini) [reply src/components/Network.tsx:102]: good catch

File: src/components/TransactionDisplay.tsx
Comment (hbulgarini) [reply src/components/TransactionDisplay.tsx:47]: i'm not agree with this, returning null is quite used and actually is documented 
https://reactjs.org/docs/conditional-rendering.html#preventing-component-from-rendering

File: src/components/Network.tsx
Comment (hbulgarini) [reply src/components/Network.tsx:102]: if you use vscode you can set to format the file during the onSave and adding the ? is one of the things that are added automatically by you.

File: src/components/TransactionDisplay.tsx
Comment (goldsteinsveta) [reply src/components/TransactionDisplay.tsx:47]: ok, I'm not correcting unless it is decided otherwise

Comment (wirednkod) [reply src/components/TransactionDisplay.tsx:47]: Just think of it this way:
Why to render (and go through the lifecycle) of a component as soon as you want to return null? - just do not render it.

Documentation you sent talks about "rare" situations (I would use it only if some internal option of the component was needed to hide it - some state change etc etc).

File: src/components/Network.tsx
Comment (wirednkod) [reply src/components/Network.tsx:102]: I do not like to let things do automatically stuff for me - I like to have control of the code :)

File: src/components/TransactionDisplay.tsx
Comment (hbulgarini) [reply src/components/TransactionDisplay.tsx:47]: i see what you mean, but anyway lifecycle components is not a such expensive process to avoid.

Comment (goldsteinsveta) [reply src/components/TransactionDisplay.tsx:47]: I haven't corrected this for now, will wait for consensus from memory mgmt gurus! Pls approve

File: src/components/TransactionDisplay.tsx
Comment (tomusdrw) [inline src/components/TransactionDisplay.tsx:52]:
```suggestion
      <ButtonSwitchMode color="primary"> Receipt</ButtonSwitchMode>
```

File: src/components/TransactionStatus.tsx
Comment (tomusdrw) [inline src/components/TransactionStatus.tsx:131]:
```suggestion
        label: 'Finalise block',
```

File: src/components/TransactionStatus.tsx
Comment (tomusdrw) [inline src/components/TransactionStatus.tsx:136]:
```suggestion
        label: 'Relay block',
```
On first impression I'd prefer `Relay` rather than `Bridge`, cause so far we've never used `bridge` as a verb. Although I know it's a bit technical.

Comment (tomusdrw) [reply src/components/TransactionStatus.tsx:136]:
```suggestion
        label: 'Receive block',
```

I just had another idea and I think we can call this "Receive block", since the `chainType` is target, wdyt?

File: src/components/TransactionStatus.tsx
Comment (tomusdrw) [inline src/components/TransactionStatus.tsx:141]:
```suggestion
        label: 'Deliver message',
```
I'd keep this in-line with other messages (`verb + what`).

File: src/components/TransactionStatus.tsx
Comment (tomusdrw) [inline src/components/TransactionStatus.tsx:142]: Is it actually useful to have `messageNonce` for anything? Maybe let's make it a target chain block in which the message was delivered instead?

File: src/components/TransactionStatus.tsx
Comment (tomusdrw) [inline src/components/TransactionStatus.tsx:147]:
```suggestion
        label: 'Finalise message',
```
It's clear it's target from `chainType` imho.

File: src/components/TransactionStatus.tsx
Comment (tomusdrw) [inline src/components/TransactionStatus.tsx:152]:
```suggestion
        label: 'Confirm delivery',
```

File: src/components/TransactionDisplay.tsx
Comment (goldsteinsveta) [reply src/components/TransactionDisplay.tsx:47]: will be resolved in https://github.com/paritytech/parity-bridges-ui/issues/108

File: src/components/TransactionStatus.tsx
Comment (goldsteinsveta) [reply src/components/TransactionStatus.tsx:136]: I was torn here, relay sounds good

File: src/components/TransactionStatus.tsx
Comment (goldsteinsveta) [reply src/components/TransactionStatus.tsx:141]: makes sense

File: src/components/TransactionStatus.tsx
Comment (goldsteinsveta) [reply src/components/TransactionStatus.tsx:142]: It's kind of is. It's useful now with current sidebar where you see the number of `outboundLanes.totalMessages` that is less than `transaction.messageNonce` when transaction is included in source block. It's like a countdown to your message nonce. Pending messages value on the outbound lane does not return the number of pending messages either. It's a patch on the UI side for the values that do not return what they suppose to

Comment (goldsteinsveta) [reply src/components/TransactionStatus.tsx:143]: Mb this makes sense only for debugging. I've added a static string to finalised on target step. Will need to refactor to pull that value. Or perhaps @hbulgarini can help me with making TransactionStatus do what I want it to do? I.e. replace `'NNN'` with actual number of finalzed block on target

**Review by tomusdrw — APPROVED** [review]

**Comment by goldsteinsveta** [comment]
thank you @tomusdrw

**Pull request merged**
